Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -3125,6 +3134,22 @@ private static bool HandleKeywords(Material destMat, string keyword)
 #if AT_USE_SPLINES
     PinnedList<float3> m_tmpFloat3 = new PinnedList<float3>();
 #endif
+    
+#if UNITY_EDITOR
+    [NonSerialized]
+    private List<RenderTexture> CreatedRenderTextures = new List<RenderTexture>();
+
+    public void AddCreatedRenderTexture(RenderTexture renderTexture) {
+        // Make sure the list doesn't grow infinitely with already released references:
+        for (int i = CreatedRenderTextures.Count - 1; i >= 0; i--) {
+            if (!CreatedRenderTextures[i].IsCreated()) {
+                CreatedRenderTextures.RemoveAt(i);
```

Comment:
I think we can simply use a hashset here, since `RemoveAt()` may reorder the elements in every iteration of the loop.

Reply:
Why use a HashSet? We don't do any lookups on this.

Reply:
because `RemoveAt()` may reorder the elements, which is an O(n) operation.
Using `HashSet` won't reorder the elements when we remove one of them, and it's an O(1) operation.

![image](https://user-images.githubusercontent.com/71803280/235425647-dfa2bcc2-bd5f-49d5-afb2-17bd2e87320f.png)

Taken from [this article](https://www.c-sharpcorner.com/UploadFile/0f68f2/comparative-analysis-of-list-hashset-and-sortedset/)

Reply:
Yes but I'm iterating backwards, which makes this much more efficient. Iterating over a HashSet is slower and I can't remove while I'm iterating because I'd need to use foreach and would need to construct another list to remove things afterwards. All this adds complexity for no gain.

Reply:
> Yes but I'm iterating backwards

Right, but we have cases where we are not removing the last elements (the `if` part above),  so removing an element might still reorder the elements with higher index in the list which were not removed before.

> I'd need to use foreach and would need to construct another list

This is true. How many RenderTextures that we expect to use here ? And how many of them are expected to return `IsCreated()` to true/false ?

Reply:
There could be some reordering, I wouldn't expect to have more than 10-20 textures so there'd be no noticeable performance benefit.

Reply:
Ok, lets merge this as is for now.